### PR TITLE
feat(transfer): allow non-cosmos-sdk AccAddress in final receiver for forwarded packets

### PR DESCRIPTION
**Pull request by gjermundgaraba**
## Description

Ref: #6696 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/transfer/keeper/relay_forwarding_test.go [hunk 1]
```diff
@@ -223,6 +223,90 @@ func (suite *KeeperTestSuite) TestSuccessfulForward() {
...
+			C: no assertions as we don't have a non-cosmos chain to test with, so this would fail here
+	*/
+
+	amount := sdkmath.NewInt(100)
+
+	path1 := ibctesting.NewTransferPath(suite.chainA, suite.chainB)
+	path1.Setup()
+
+	path2 := ibctesting.NewTransferPath(suite.chainB, suite.chainC)
+	path2.Setup()
```
Comment (crodriguezvega) [inline modules/apps/transfer/keeper/relay_forwarding_test.go:250]: You could do this like in the other tests:

```suggestion
	pathAtoB, pathBtoC := suite.setupForwardingPaths()
```

**Review by crodriguezvega — APPROVED** [review]
Excellent!

File: modules/apps/transfer/keeper/relay_forwarding_test.go (hunk 1 above)
Comment (gjermundgaraba) [reply modules/apps/transfer/keeper/relay_forwarding_test.go:250]: Oh yeah, that is much better obviously. Forgot about that one. Will fix! :)

**Comment by elix1er** [comment]
is this to support evm type addresses ? A+

**Comment by damiannolan** [comment]
> is this to support evm type addresses ? A+

Yep! Essentially avoids any assumptions about the final receiver field in the ibc transfer msg being cosmos-sdk specific, such that addresses for other ecosystems can be provided in the string field 👍🏻

**Pull request merged**